src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -410,16 +411,15 @@ static Slf4jLogMessageContext parseSlf4jContextFromLogMessage(String logMessage)
     public void addLogMessageConsumer(Consumer<String> consumer) {
         checkNotNull(consumer, "consumer parameter is NULL!");
         addOutPutStream(new OutputStream() {
-            StringBuilder lineBuilder = new StringBuilder();
+            private ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
 
             @Override
             public void write(int chr) {
                 try {
-                    lineBuilder.append((char) chr);
-                    Matcher matcher = LOG_MESSAGE_PATTERN.matcher(lineBuilder.toString());
-                    if (matcher.matches()) {
-                        consumer.accept(matcher.group(1));
-                        lineBuilder = new StringBuilder();
+                    outputStream.write(chr);
+                    if ((chr == '\n')) {
+                        consumer.accept(outputStream.toString());
```

Comment:
should we explicitly provide the encoding here?

Reply:
default charset is good for me

Reply:
it makes sense to introduce the ability to rely on default charset or use a custom one